src/Integration.Vsix/SonarLintDaemon.cs

```diff
@@ -0,0 +1,140 @@
+{
+    class SonarLintDaemon
+    {
+        public const string daemonVersion = "2.12.0.492";
+        private const string uriFormat = "https://repox.sonarsource.com/sonarsource-public-builds/org/sonarsource/sonarlint/core/sonarlint-daemon/{0}/sonarlint-daemon-{0}-windows.zip";
+        private readonly string version;
+        private readonly string tmpPath;
+        private readonly string storagePath;
+
+        private Process process;
+
+        public SonarLintDaemon() : this(daemonVersion, Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), Path.GetTempPath())
+        {
+        }
+
+        public SonarLintDaemon(string version, string storagePath, string tmpPath)
+        { 
+            this.version = version;
+            this.tmpPath = tmpPath;
+            this.storagePath = storagePath;
+        }
+
+        public void Install()
+        {
+            Download();
+            Unzip();
+        }
+
+        public bool IsRunning()
+        {
+            if (process == null)
+            {
+                return false;
+            }
+            return !process.HasExited;
+        }
+
+        public void Start()
+        {
+            if (IsRunning())
+            {
+                throw new InvalidOperationException("Process already running");
+            }
+            if (!IsInstalled())
+            {
+                throw new InvalidOperationException("Daemon is not installed");
+            }
+
+            if (process != null)
+            {
+                process.Close();
+            }
+
+            process = new Process();
+            process.StartInfo.UseShellExecute = false;
+            process.StartInfo.FileName = GetExePath();
+            process.StartInfo.Arguments = GetCmdArgs();
+            process.StartInfo.CreateNoWindow = true;
+            process.Start();
+        }
+
+        public void Stop()
+        {
+            if (!IsRunning())
+            {
+                return;
+                // throw exception?
+            }
+            process.Kill();
+            process.WaitForExit();
+        }
+
+        public bool IsInstalled()
+        {
+            return Directory.Exists(GetInstallationPath()) && File.Exists(GetExePath());
+        }
+
+        #region private methods
+
+        private void Download()
+        {
+            string uri = string.Format(uriFormat, version);
+            using (var client = new WebClient())
+            {
+                client.DownloadFile(uri, GetZipFilePath());
+            }
+        }
+
+        private void Unzip()
+        {
+            if (Directory.Exists(GetInstallationPath()))
+            {
+                Directory.Delete(GetInstallationPath(), true);
+            }
+            ZipFile.ExtractToDirectory(GetZipFilePath(), storagePath);
+        }
+
+        private string GetInstallationPath()
+        {
+            return Path.Combine(storagePath, "sonarlint-daemon-" + version + "-windows");
```

Comment:
This could possibly be a bit easier to read:

    return Path.Combine(storagePath, $"sonarlint-daemon-{version}-windows");